### ISSUE TITLE
Classpath Incomplete Warning with Eclipse-generated project

### ISSUE THREAD

**Issue by bbeheydt** [redhat-developer/vscode-java#504]
I have a lot of projects created through Eclipse that work fine. They have buildpaths and classpaths very similar to a project I'm working on currently. I created the project, made some classes, and then started implementing stuff in VSCode when the Classpath Incomplete warning popped up. The .classpath file is in the project folder, along with the .project file. I have the default src folder with a package within it that has all the source code. Everything was setup using Eclipse.

##### Environment
- Operating System: Win10.something
- JDK version: 1.8.0_144
- Visual Studio Code version: 1.22.1
- Java extension version: 0.22.0

##### Steps To Reproduce
1. Create a lot of projects in Eclipse.
2. Write all of the source in VSCode.
3. Have one project created in Eclipse repeatedly throw Classpath Incomplete warning in VSCode.

##### Current Result
ClassPath Incomplete warning thrown
##### Expected Result
No warnings being thrown
##### Additional Informations
If the warnings could include a diagnostic in a future version, that may help a lot.
And sorry, I can't provide a video because I can't post my source online anywhere.

**Comment by fbricon** [comment]
Can you share the content of your .project and .classpath files?

**Comment by bbeheydt** [comment]
Just realized that didn't work. Give me a sec.

**Comment by bbeheydt** [comment]
![image](https://user-images.githubusercontent.com/36484758/38701046-e2911502-3e6a-11e8-8b5e-9404e435371c.png)

![image](https://user-images.githubusercontent.com/36484758/38701099-fa80c5c2-3e6a-11e8-9fea-127e4c9da218.png)

**Comment by fbricon** [comment]
How are structured your projects? all under the same root directory? Do you open that root directory in vscode?

**Comment by bbeheydt** [comment]
Yes. All projects are in one directory and no library referenced in the classpath exists outside of that directory. I have other projects within that directory and that reference the same libraries but seem to work just fine. Here are the .classpath and .project files for a project that is logically the same as the problem project.

![image](https://user-images.githubusercontent.com/36484758/38701862-0482b380-3e6d-11e8-8d10-3cf401d9c1da.png)

![image](https://user-images.githubusercontent.com/36484758/38701894-1a61ec5c-3e6d-11e8-9de1-9bd7df4c46e4.png)

**Comment by fbricon** [comment]
The "Classpath Incomplete" Warning is displayed when the file you opened can't be referenced to a working Eclipse project. So, chances are, offending project was not imported when opening the folder.
But it's hard to understand why without a way to reproduce it.

Any chance you could zip and attach all your projects without their java sources? That'd allow us to debug the project import process and hopefully understand why it fails for a particular project.

**Comment by bbeheydt** [comment]
Unfortunately, the libraries are too large to be sent through this site. The resulting zip file is 29MB, and most of that memory comes from the libraries referenced in the classpath file. Any other ways to do this?

**Comment by fbricon** [comment]
remove the jars. You should end up with a basic skeleton of an app

**Comment by bbeheydt** [comment]
Okay, after gutting the jars from the libraries I'm left with a zip that is still slightly over size. The memory is mainly coming from a .metadata folder. Do you need that for anything? (Sorry, I'm clueless)

**Comment by fbricon** [comment]
No,  don't need . metadata

**Comment by bbeheydt** [comment]
Okay. The only gutted library in this zip file is CS2114-Support. The other libraries are stocked with their jars. The Ex*** projects are all working projects. The SpaceColonies project is the offending project. It has a TestClass.java in it with no real code. Let me know if you need anything else! Thanks!

[eclipse workspace.zip](https://github.com/redhat-developer/vscode-java/files/1905121/eclipse.workspace.zip)

**Comment by fbricon** [comment]
I don't get a warning in TestClass.java after opening SpaceColonies.
If I force the workspace compilation (shift+alt+b), I can see a bunch of error messages in the Java server output, caused by the missing jars. Please try that command and report any error you get.

**Comment by fbricon** [comment]
Oh there was also an error caused by the missing CarranoDataStructures project. Which is defined by .project and .classpath under CarranoDataStructuresLib/lib which is highly suspicious. I removed them.

**Comment by bbeheydt** [comment]
It's worth noting that everything works in eclipse. All programs function properly, all written tests pass, and no compilation/project errors are thrown. Also, the KVStore.java file mentioned above is NOT in the path that is printed. It's actually ~src/cybersecurity/KVStore.java

**Comment by bbeheydt** [comment]
Okay so I moved KVStore.java into the location that the error said it was supposed to be in. Then, I forced a compilation, and the output stated that another file was out of place. So I followed that trail (put all these files 'back' to where they were 'supposed' to be) and then it started outputting files that were delete a long time ago. All of the problem files have the file path

 '/jdt.ls-java-project/src/********' like stated above

This is strange to me since technically all of the files are in their own src folders in their own projects. They aren't all in own big src file.

**Comment by bbeheydt** [comment]
Here is the output trail I followed until I got to ~/src/lab11pre/package-info.java. This file (and in fact the project) are long gone.

[Error - 7:23:07 PM] Apr 12, 2018 7:20:27 PM Error occured while building workspace. Details: 
message: Project 'CarranoDataStructures' is missing required source folder: 'src';code: 964
message: The project cannot be built until build path errors are resolved;code: 0
message: The project was not built since the source file **/jdt.ls-java-project/src/cybersecurity/lab2/KVStore.java could not be read;code: 0**
[Error - 7:32:42 PM] Apr 12, 2018 7:20:27 PM Error occured while building workspace. Details: 
message: Project 'CarranoDataStructures' is missing required source folder: 'src';code: 964
message: The project cannot be built until build path errors are resolved;code: 0
message: The project was not built since the source file **/jdt.ls-java-project/src/lab11pre/KVListInterface.java could not be read;code: 0**
[Error - 7:33:42 PM] Apr 12, 2018 7:20:27 PM Error occured while building workspace. Details: 
message: Project 'CarranoDataStructures' is missing required source folder: 'src';code: 964
message: The project cannot be built until build path errors are resolved;code: 0
message: The import cybersecurity.KVStore cannot be resolved;code: 268435846
message: The project was not built since the source file **/jdt.ls-java-project/src/linkedlist/LList.java could not be read;code: 0**
[Error - 7:39:34 PM] Apr 12, 2018 7:20:27 PM Error occured while building workspace. Details: 
message: Project 'CarranoDataStructures' is missing required source folder: 'src';code: 964
message: The project cannot be built until build path errors are resolved;code: 0
message: The project was not built since the source file **/jdt.ls-java-project/src/lab8pre.java could not be read;code: 0**
[Error - 7:41:32 PM] Apr 12, 2018 7:20:27 PM Error occured while building workspace. Details: 
message: Project 'CarranoDataStructures' is missing required source folder: 'src';code: 964
message: The project cannot be built until build path errors are resolved;code: 0
message: The project was not built since the source file **/jdt.ls-java-project/src/lab11pre/package-info.java could not be read;code: 0**
